packages/apps/shopify-app-remix/src/server/authenticate/admin/scope/types.ts

```diff
@@ -1,14 +1,166 @@
   query: () => Promise<ScopesDetail>;
-  request: (scopes: string[]) => Promise<void>;
-  revoke: (scopes: string[]) => Promise<RevokeResponse>;
-}
 
-export interface RevokeResponse {
-  revoked: string[];
+  /**
+   * Requests the merchant to grant the provided scopes for this app on this shop
+   *
+   * **Warning:** This method performs a server-side redirect.
+   *
+   * @example
+   * <caption>Request consent from the merchant to grant the provided scopes for this app.</caption>
+   * <description>Call `scopes.request` to request optional scopes.</description>
+   * ```ts
+   * // /app/routes/app.request.tsx
+   *import type { LoaderFunctionArgs } from "@remix-run/node";
+   *import { json } from "@remix-run/node";
+   *
+   *import { authenticate } from "../shopify.server";
+   *import { AuthScopes } from "@shopify/shopify-api";
+   *export async function loader({
+   *  request,
+   *}: LoaderFunctionArgs) {
+   *  const { scopes } = await authenticate.admin(request);
+   *
+   *  const body = await request.formData();
+   *  const optionalScopesData = body.getAll("scopes") as string[];
+   *  const optionalScopes = new AuthScopes(optionalScopesData);
+   *
+   *  await scopes.request(optionalScopes.toArray());
+   *  // nothing to grant
+   *  return json({});
+   *}
+   * ```
+   * ```ts
+   * // /app/shopify.server.ts
+   * import { LATEST_API_VERSION, shopifyApp } from "@shopify/shopify-app-remix/server";
+   *
+   * const shopify = shopifyApp({
+   *   // ...etc
+   * });
+   * export default shopify;
+   * export const authenticate = shopify.authenticate;
+   * ```
+   */
+  request: (scopes: Scope[]) => Promise<void>;
+
+  /**
+   * Revokes the provided scopes from this app on this shop
+   *
+   * **Warning:** This method throws an [error](https://shopify.dev/docs/api/admin-graphql/unstable/objects/AppRevokeAccessScopesAppRevokeScopeError) if the provided optional scopes contains a required scope. 
+   *
+   * @example
+   * <caption>Revoke optional scopes.</caption>
+   * <description>Call `scopes.revoke` to revoke optional scopes.</description>
+   * ```ts
+   * // /app/routes/app.revoke.tsx
+   *import type { LoaderFunctionArgs } from "@remix-run/node";
+   *import { json } from "@remix-run/node";
+   *import { authenticate } from "../shopify.server";
+   *import { AuthScopes } from "@shopify/shopify-api";
+   *
+   *export async function loader({
+   *  request,
+   *}: LoaderFunctionArgs) {
+   *  const { scopes } = await authenticate.admin(request);
+   *
+   *  const body = await request.formData();
+   *  const scopesToRevokeData = body.getAll("scopes") as string[];
+   *  const scopesToRevoke = new AuthScopes(scopesToRevokeData);
+   *
+   *  try {
+   *    const revokedResponse = await scopes.revoke(scopesToRevoke.toArray());
+   *    return json(revokedResponse);
+   *  } catch (e) {
+   *    console.log(JSON.stringify(e));
+   *    return json({});
+   *  }
+   *}
+   * ```
+   * ```ts
+   * // /app/shopify.server.ts
```

Comment:
As it stands, I don't know what value these app/shopify.server.ts files are providing. (What are they saying that is different from the `shopify.server.ts` they get at app install? Are the first tabs sufficient to show how to set up & access the API?)